core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -174,9 +174,13 @@ trait Endpoint[A] { self =>
       }
 
       final def apply(input: Input): Endpoint.Result[O] = self(input) match {
-        case a @ EndpointResult.Matched(_, _) => other(a.rem) match {
-          case b @ EndpointResult.Matched(_, _) =>
-            EndpointResult.Matched(b.rem, a.out.liftToTry.product(b.out.liftToTry).flatMapF(this))
+        case a @ EndpointResult.Matched(_, _, _) => other(a.rem) match {
+          case b @ EndpointResult.Matched(_, _, _) =>
+            EndpointResult.Matched(
+              b.rem,
+              a.trc.concat(b.trc),
```

Comment:
what if we move this `concat` to `private val trc = a.trc.concat(b.trc)` of the endpoint to generate it only once?

Reply:
Sorry, I'm not sure this is possible. Neither `a.trc` nor `b.trc` are available until we run an endpoint (`Endpoint.apply`).

Reply:
Ah, yeah

nvm then.